### PR TITLE
Added Hard Signum Bunny

### PR DESCRIPTION
**Pull request by Atman-Kar**
# Added ```HardSignBunny()``` class 

Applies a hard signum function to all the weights and biases of the model and spit out new model.

### CODE REVIEW

File: src/bunnies/abstract.py [hunk 1]
```diff
@@ -4,31 +4,32 @@
...
 
-    def swap_layers(self, model, *args, **kwargs):
-        
         list_model = list(model.children())
         for idx, layer in enumerate(list_model):
             try:
                 layer.weight.data = self.binarize(layer.weight.data)
             except:
-                print(f"Cannot binarize")
+                if verbose:
```
Comment (hades-rp2010) [inline src/bunnies/abstract.py:26]: There's no place where `verbose` is taken as an argument.  
A better suggestion: `if kwargs.get(verbose, False):`  
What do you think?

File: src/bunnies/random_bunny.py [hunk 2]
```diff
@@ -8,5 +8,5 @@ def __init__(self):
         super().__init__()
 
     
-    def binarize(self, parameter):
+    def binarize(self, parameter) -> torch.Tensor:
```
Comment (hades-rp2010) [inline src/bunnies/random_bunny.py]: This `-> torch.Tensor` would be better placed in the abstract class no?

**Review by hades-rp2010 — CHANGES_REQUESTED** [review]

File: src/bunnies/abstract.py (hunk 1 above)
Comment (Atman-Kar) [reply src/bunnies/abstract.py:26]: Done

File: src/bunnies/random_bunny.py (hunk 2 above)
Comment (Atman-Kar) [reply src/bunnies/random_bunny.py]: Done

File: src/bunnies/abstract.py (hunk 1 above)
Comment (hades-rp2010) [reply src/bunnies/abstract.py:26]: Cool

**Review by hades-rp2010 — APPROVED** [review]
Merging this. I dont see any probs right now

**Pull request merged**